core/common/src/main/java/alluxio/Configuration.java

```diff
@@ -432,38 +428,34 @@ public static void loadClusterDefault(InetSocketAddress address) throws AlluxioS
       }
       LOG.info("Alluxio client (version {}) is trying to bootstrap-connect with {}",
           RuntimeConstants.VERSION, address);
-      // A plain socket transport to bootstrap
-      TSocket socket = ThriftUtils.createThriftSocket(address);
-      TTransport bootstrapTransport = new BootstrapClientTransport(socket);
-      TProtocol protocol = ThriftUtils.createThriftProtocol(bootstrapTransport,
-          Constants.META_MASTER_CLIENT_SERVICE_NAME);
-      List<ConfigProperty> clusterConfig;
+
+      GrpcChannel channel = null;
+      List<alluxio.grpc.ConfigProperty> clusterConfig = null;
+
       try {
-        bootstrapTransport.open();
-        // We didn't use RetryHandlingMetaMasterClient because it inherits AbstractClient,
-        // and AbstractClient uses Configuration.loadClusterDefault inside.
-        MetaMasterClientService.Client client = new MetaMasterClientService.Client(protocol);
-        // The credential configuration properties use displayValue
-        clusterConfig = client.getConfiguration(new GetConfigurationTOptions().setRawValue(true))
-            .getConfigList()
-            .stream()
-            .map(ConfigProperty::fromThrift)
-            .collect(Collectors.toList());
-      } catch (TException e) {
+        // TODO(ggezer) review grpc channel initialization
+        channel = GrpcChannelBuilder.forAddress("localhost", 50051).usePlaintext(true).build();
+        MetaMasterClientServiceGrpc.MetaMasterClientServiceBlockingStub client =
+            MetaMasterClientServiceGrpc.newBlockingStub(channel);
+        clusterConfig =
+            client.getConfiguration(GetConfigurationPOptions.newBuilder().setRawValue(true).build())
+                .getConfigsList();
+      } catch (io.grpc.StatusRuntimeException e) {
         throw new UnavailableException(String.format(
             "Failed to handshake with master %s to load cluster default configuration values",
             address), e);
       } finally {
-        bootstrapTransport.close();
+        channel.shutdown();
       }
+
       // merge conf returned by master as the cluster default into Configuration
       Properties clusterProps = new Properties();
       for (ConfigProperty property : clusterConfig) {
         String name = property.getName();
         // TODO(binfan): support propagating unsetting properties from master
-        if (PropertyKey.isValid(name) && property.getValue() != null) {
+        if (PropertyKey.isValid(name) && property.hasValue()) {
           PropertyKey key = PropertyKey.fromString(name);
-          if (!key.getScope().contains(Scope.CLIENT)) {
+          if (!GrpcUtils.contains(key.getScope(), Scope.CLIENT)) {
```

Comment:
why move the contains logic to `GrpcUtils`?

Reply:
That logic was handled in the wire type. I removed the wire type and that logic needs to go somewhere. This logic could indeed be implemented agnostic of enum type.